static/js/auth.ts

```diff
@@ -82,123 +82,141 @@ export function request_teacher_account() {
 
 // *** User forms ***
 
-$('form#signup').submit(function(e) {
-  e.preventDefault();
-  $.ajax ({
-        type: 'POST',
-        url: '/auth/signup',
-        // This should do the magic to convert to a correct JSON object
-        data: convertFormJSON($(this)),
-        contentType: 'application/json; charset=utf-8'
-      }).done (function () {
-        afterLogin({"first_time": true});
-      }).fail (function (response) {
-        modal.alert(response.responseText, 3000, true);
-      });
-});
+export function initializeFormSubmits() {
+  $('form#signup').submit(function(e) {
```

Comment:
Since we already are changing so much of the code, can we change the `submit` function and instead call `on`? `submit` and similar shorthand event handlers were deprecated since JQuery 3.3.